src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisWebhookService.java

```diff
@@ -74,6 +77,57 @@ public PyrisWebhookService(PyrisConnectorService pyrisConnectorService, PyrisJob
         this.lectureRepository = lectureRepository;
     }
 
+    /**
+     * adds the transcription to the vector database in Pyris
+     *
+     * @param transcriptions The transcription that got Updated
+     * @param course         The course of the transcriptions
+     * @param lecture        The lecture of the transcriptions
+     * @param lectureUnit    The lecture unit of the transcriptions
+     * @return jobToken if the job was created else null
+     */
+    public String addTranscriptionsToPyrisDB(Set<LectureTranscription> transcriptions, Course course, Lecture lecture, LectureUnit lectureUnit) {
+        if (transcriptions == null || transcriptions.isEmpty()) {
+            throw new IllegalArgumentException("Transcriptions cannot be empty");
+        }
+
+        if (!lectureIngestionEnabled(course)) {
+            return null;
+        }
+
+        transcriptions.forEach(transcription -> {
+            if (transcription.getLecture() == null) {
+                throw new IllegalArgumentException("Transcription must be associated with a lecture");
+            }
+            else if (!transcription.getLecture().equals(lecture)) {
+                throw new IllegalArgumentException("All transcriptions must be associated with the same lecture");
+            }
+        });
+
+        List<PyrisTranscriptionIngestionWebhookDTO> pyrisTranscriptionIngestionWebhookDTOs = transcriptions.stream()
+                .map(transcription -> new PyrisTranscriptionIngestionWebhookDTO(transcription, lecture.getId(), lecture.getTitle(), course.getId(), course.getTitle(),
+                        course.getDescription()))
+                .toList();
+
+        return executeTranscriptionAdditionWebhook(pyrisTranscriptionIngestionWebhookDTOs, course, lecture, lectureUnit);
+    }
+
+    /**
+     * executes executeTranscriptionAdditionWebhook to insert the transcription into the vector database of Pyris
```

Comment:
this detail should not be part of the comment. Artemis does not care about the infrastructure of pyris